Mathlib/Algebra/Module/Injective.lean

```diff
@@ -465,13 +464,69 @@ protected theorem injective (h : Module.Baer R Q) : Module.Injective R Q :=
 set_option linter.uppercaseLean3 false in
 #align module.Baer.injective Module.Baer.injective
 
-protected theorem of_injective (inj : Module.Injective R Q) : Module.Baer R Q := fun I g ↦
-  let ⟨g', hg'⟩ := inj.1 (ULift.{max u v} I) (ULift.{max u v} R)
-    (ULift.moduleEquiv.symm.toLinearMap ∘ₗ I.subtype ∘ₗ ULift.moduleEquiv.toLinearMap)
-    (fun a b h ↦ by aesop) (g ∘ₗ ULift.moduleEquiv.toLinearMap)
-  ⟨g' ∘ₗ ULift.moduleEquiv.symm.toLinearMap, by aesop⟩
+protected theorem of_injective (inj : Module.Injective R Q) : Module.Baer R Q := by
+  intro I g
```

Comment:
This proof has got a lot longer, and isn't UnivLE supposed to make things easier? What's going on here?

Reply:
`UnivLE` make proof easier in the sense that less universe variable need to be made explicit. The length is longer because of `I` can't be lifted using `ULift` any more but need to use `Shrink I` whose instances needs to be manually introduced via injection `I \equiv Shrink I`

Reply:
See ![this](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Universe.20trouble.20when.20using.20.60.60.60Module.2EInjective.60.60.60/near/403006687) problem

Reply:
I think you can put AddCommGroup and Module structures on `Shrink I` just like [ULift.module](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/ULift.html#ULift.module) and then construct a LinearEquiv between `I` and `Shrink I` in parallel to [ULift.moduleEquiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/ULift.html#ULift.moduleEquiv), which you can then compose with the linear maps involved here.

It's unfortunate that we have to duplicate the ULift API; if [lean4#2297](https://github.com/leanprover/lean4/pull/2297) could be solved we wouldn't need to open the UnivLE business ... (if you want to see it happen please upvote)

Reply:
I did this in [634f566](https://github.com/leanprover-community/mathlib4/pull/8905/commits/634f56646d69d5e68a4b821f9430167479b05786), but I think this should be a separate PR

Reply:
#8998

Reply:
Now the proof of this lemma is about the same size!